aws_lambda_powertools/utilities/data_classes/kinesis_firehose_event.py

```diff
@@ -1,9 +1,123 @@
+    - https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html
+    """
+
+    partition_keys: Optional[Dict[str, str]]
+
+    def asdict(self) -> Optional[Dict]:
+        if self.partition_keys is not None:
+            return {"partitionKeys": self.partition_keys}
+        return None
+
+
+@dataclass
+class KinesisFirehoseResponseRecord:
+    """Record in Kinesis Data Firehose event
+
+    Documentation:
+    --------------
+    - https://docs.aws.amazon.com/firehose/latest/dev/data-transformation.html
+    """
+
+    """Record ID; uniquely identifies this record within the current batch"""
+    record_id: str
+    """processing result, supported value: Ok, Dropped, ProcessingFailed"""
+    result: Literal["Ok", "Dropped", "ProcessingFailed"]
+    """data blob, base64-encoded, optional at init. Allows pass in base64-encoded data directly or
+            use either function like `data_from_text`, `data_from_json` to populate data"""
+    data: Optional[str] = None
+    """Optional: metadata associated with this record; present only when Kinesis Stream is source"""
+    metadata: Optional[KinesisFirehoseResponseRecordMetadata] = None
+    """Json data for caching json.dump result"""
+    _json_data: Optional[Any] = None
+    json_serializer: Callable = json.dumps
+    json_deserializer: Callable = json.loads
+
+    def data_from_byte(self, data: bytes):
+        """Populate data field using a byte like data"""
+        self.data = base64.b64encode(data).decode("utf-8")
+
+    def data_from_text(self, data: str):
+        """Populate data field using a string like data"""
+        self.data_from_byte(data.encode("utf-8"))
+
+    def data_from_json(self, data: Any):
+        """Populate data field using any structure that could be converted to json"""
+        self.data_from_text(data=self.json_serializer(data))
+
+    def asdict(self) -> Dict:
+        r: Dict[str, Any] = {
+            "recordId": self.record_id,
+            "result": self.result,
+            "data": self.data,
+        }
+        if self.metadata:
+            r["metadata"] = self.metadata.asdict()
+        return r
+
+    @property
+    def data_as_bytes(self) -> Optional[bytes]:
+        """Decoded base64-encoded data as bytes"""
+        if not self.data:
+            return None
+        return base64.b64decode(self.data)
+
+    @property
+    def data_as_text(self) -> Optional[str]:
+        """Decoded base64-encoded data as text"""
+        if not self.data_as_bytes:
+            return None
+        return self.data_as_bytes.decode("utf-8")
+
+    @property
+    def data_as_json(self) -> Optional[Dict]:
+        """Decoded base64-encoded data loaded to json"""
+        if not self.data_as_text:
+            return None
+        if self._json_data is None:
+            self._json_data = self.json_deserializer(self.data_as_text)
+        return self._json_data
+
+
+@dataclass
+class KinesisFirehoseResponse:
+    """Kinesis Data Firehose event
```

Comment:
May want to change this to "Kinesis Data Firehose response object" so it's not confused with the event source object.

Reply:
+1